### PR TITLE
Disable HMM feature of Jieba

### LINKED ISSUE #136: Disable HMM feature of Jieba
Today, we are using the Hidden Markov Model algorithm (HMM) provided by the [`cut` method](https://docs.rs/jieba-rs/0.6.6/jieba_rs/struct.Jieba.html#method.cut) of Jieba to segment unknown Chinese words in the [Chinese segmenter](https://github.com/meilisearch/charabia/blob/dac6b422e43b666cc232430533e61a50abdd3341/src/segmenter/chinese.rs#L14).

#### drawback
Following [the subdiscussion](https://github.com/meilisearch/product/discussions/503#discussioncomment-3456196) in the official discussion about Chinese support in Meilisearch, it seems that the HMM feature of Jieba is not relevant in the context of a search engine. This feature creates longer words and inconsistencies in the segmentation, which reduces the recall of Meilisearch without significantly raising the precision.

#### enhancement

Deactivate the HMM feature in Chinese segmentation.

#### Files expected to be modified
- [/src/segmenter/chinese.rs](https://github.com/meilisearch/charabia/blob/dac6b422e43b666cc232430533e61a50abdd3341/src/segmenter/chinese.rs)

#### Misc
related to [product#503](https://github.com/meilisearch/product/discussions/503)

> Hey! 👋 
Before starting any implementation, make sure that you read the [CONTRIBUTING.md](https://github.com/meilisearch/charabia/blob/main/CONTRIBUTING.md#contributing) file.
In addition to the recurrent rules, you can find some guides to easily implement a `Segmenter` or a `Normalizer`.
Thanks a lot for your Contribution! 🤝

### PR DESCRIPTION
**Pull request by aniketh-varma**
# Pull Request

## What does this PR do?
Fixes #136 

## PR checklist
Please check if your PR fulfills the following requirements:
- [x] Does this PR fix an existing issue?
- [x] Have you read the contributing guidelines?
- [x] Have you made sure that the title is accurate and descriptive of the changes?

Thank you so much for contributing to Meilisearch!

### CODE REVIEW

File: src/segmenter/chinese.rs
Comment (ManyTheFish) [inline src/segmenter/chinese.rs:9]:
```suggestion
/// without using HMM feature.
```

**Review by ManyTheFish — CHANGES_REQUESTED** [review]
Hello @aniketh-varma!
Thanks for your pull request, I requested a small change in it.
I will merge the PR after you made the change. 😄

File: src/segmenter/chinese.rs
Comment (aniketh-varma) [reply src/segmenter/chinese.rs:9]: Sure, I will do the changes right away.

**Review by ManyTheFish — APPROVED** [review]
Great! Thank you for your contribution!

bors merge

**Comment by ManyTheFish** [comment]
Hello @aniketh-varma,

The tests are failing, I suggest running `cargo test chinese` locally and modify the tests to make them work,

Thanks!

**Comment by ManyTheFish** [comment]
Re @aniketh-varma,
Your work is conflicting with a [PR that has been recently merged](https://github.com/meilisearch/charabia/pull/143). You will have to make a small rebase before being able to merge your PR.

**Comment by aniketh-varma** [comment]
> Hello @aniketh-varma, the tests are failing and your branch is not up to date with the main branch, If you want to ensure the tests pass, you have to run `cargo test` locally.
> 
> here is the checklist to make the PR reviewable:
> 
> * [ ]  the command `cargo test` is successful
> * [ ]  the branch is up to date with `main`
> 
> I close this PR for now, don't hesitate to reopen it when you are ready! thanks!

Sure @ManyTheFish, I will make the changes in the tests and update the branch as well

**Pull request closed without merging**
